hibernate-core/src/main/java/org/hibernate/tool/schema/extract/internal/InformationExtractorJdbcDatabaseMetaDataImpl.java

```diff
@@ -282,7 +282,7 @@ private TableInformation locateTableInNamespace(
 
 		if ( extractionContext.getJdbcEnvironment().getNameQualifierSupport().supportsCatalogs() ) {
 			if ( catalog == null ) {
-				catalogFilter = "";
+				catalogFilter = null;
```

Comment:
Our problem arose with Microsoft SQL Server using JDBC driver "jtds 1.2.7".

Why should we restrict to those without any catalog when `catalog` is null? If the search for a table should be restricted to those without catalog, then a value of `""` should already be in `catalog`, don't you think?